### ISSUE TITLE
Closing the tab using the mousewheel on any file change causes Atom to crash

### ISSUE THREAD

**Issue by xanlaeron** [atom/atom#15288]
### Prerequisites

-   [x] Put an X between the brackets on this line if you have done all of the following:
    -   Reproduced the problem in Safe Mode: <http://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode>
    -   Followed all applicable steps in the debugging guide: <http://flight-manual.atom.io/hacking-atom/sections/debugging/>
    -   Checked the FAQs on the message board for common solutions: <https://discuss.atom.io/c/faq>
    -   Checked that your issue isn't already filed: <https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom>
    -   Checked that there is not already an Atom package that provides the described functionality: <https://atom.io/packages>

### Description

Generally, closing the tabs by clicking the mousewheel doesn't have any issues at all **as long as the tabs being closed are of files that have already been saved**. But if the file is **not** saved, Atom crashes right after displaying the dialog box that asks the user whether to save or discard the changes made.

### Steps to Reproduce

The three examples enumerated below -- _renaming_, _modifying_, and _deleting a file_, all have the same end result. A dialog box will appear and Atom hangs and crashes.

#### Renaming a file

1.  Create a new file.

```sh
touch test.sh
```

2.  Open file in Atom using `atom --safe`.
3.  Add text.

```sh
#!/usr/bin/env bash
echo "Hello World!"
```

4.  Save the file but **leave the tab open**.
5.  In the terminal, do:

```sh
mv test.sh hello.sh
```

Atom's tab will display a dot that indicates that a change has occurred, which in this case is the **filename**. 

6.  Close the tab using your **mousewheel**. A dialog box would open asking if you would like to save or discard the changes.

#### Modifying a file

1.  Create a new file.

```sh
touch test.txt
```

2.  Open file in Atom using `atom --safe`.
3.  Add text.

```txt
Lorem ipsum dolor sit amet, consectetur adipiscing elit. 
```

4.  Save the file but **leave the tab open**.
5.  Delete the contents and replace it with.

```txt
The quick brown fox jumps over the lazy dog.
```

Atom's tab will display a dot that indicates that a change has occurred, which in this case is the **content**.

6.  Close the tab using your **mousewheel**. A dialog box would open asking if you would like to save or discard the changes.

#### Deleting a file

1.  Open Atom using `atom --safe`.
2.  Open `test.txt` from the previous example.
3.  In the terminal, do:

```sh
rm test.txt
```

Atom's tab will display a dot that indicates that a change has occurred, which in this case is the file's **existence**.

4.  Close the tab using your **mousewheel**. A dialog box would open asking if you would like to save or discard the changes.

**Expected behavior:** 
Upon using the mousewheel to close the tab **_(for modified, renamed, or deleted files)_** a dialog box would appear and ask if the user would like to save or discard without hanging/crashing.

**Actual behavior:**
Getting the dialog box below only appears after opening a terminal and doing `killall atom` as Atom hangs and no longer responds to any user input and I'm also unable to do anything unless I do so. Clicking or right-clicking anywhere for that matter is futile (I'm using dual screens).

The other options I have would be `ctrl + alt + backspace` to restart X11 and `alt + sysrq + b` to force reboot.

[![Screenshot_from_2017-08-13_15-15-25.png](https://s26.postimg.org/dq9uyuxvt/Screenshot_from_2017-08-13_15-15-25.png)](https://postimg.org/image/wvd48mcjp/)

**Reproduces how often:**
99.9% in `atom --safe` mode

### Versions

-   `atom --version` output:

```sh
Atom    : 1.19.0
Electron: 1.6.9
Chrome  : 56.0.2924.87
Node    : 7.4.0
```

-   `apm --version` output:

```sh
apm  1.18.2
npm  3.10.10
node 6.9.5 x64
python 2.7.12
git 2.7.4
```

-   OS / OS Version:

```sh
Linux Mint 18.2
```

### Additional Information

```sh
Kernel: 4.8.0-58-generic x86_64 (64 bit gcc: 5.4.0)
Desktop: Cinnamon 3.4.6 (Gtk 3.18.9-1ubuntu3.3)
CPU: Quad core Intel Core i7-4770S
GPU: NVIDIA GK104 [GeForce GTX 760]
Memory: 3244MiB / 11954MiB
```

`node -v` output:
```sh
v8.2.1
```

`npm -v` output:
```sh
5.3.0
```

I have experienced this issue with Atom 1.18.0 as well but I ~~ignored~~ didn't report it as I assumed that the upcoming iteration 1.19.0 (at that time) may or will address this issue.

Please let me know if there's anything else I can do _(e.g., terminal output that you might need)_ to help you in figuring out how to debug or fix this issue.

**Comment by 50Wliu** [comment]
Seems related to #15197, but that issue only started occurring with Atom 1.19.0.

**Comment by xanlaeron** [comment]
>...but that issue only started occurring with Atom 1.19.0.

@50Wliu I just installed v1.19.0 yesterday. I was experiencing the same thing the day before I installed the new version.

**Comment by xanlaeron** [comment]
This issue has been resolved in #15197 and is no longer present in Atom 1.19.3.
